packages/storage/src/providers/AWSS3ProviderManagedUpload.ts

```diff
@@ -74,51 +74,56 @@ export class AWSS3ProviderManagedUpload {
-			return this.s3client.send(putObjectCommand);
-		} else {
-			// Step 1: Initiate the multi part upload
-			const uploadId = await this.createMultiPartUpload();
-
-			// Step 2: Upload chunks in parallel as requested
-			const numberOfPartsToUpload = Math.ceil(
-				this.totalBytesToUpload / this.minPartSize
-			);
-
-			const parts: Part[] = this.createParts();
-			for (
-				let start = 0;
-				start < numberOfPartsToUpload;
-				start += this.queueSize
-			) {
-				/** This first block will try to cancel the upload if the cancel
-				 *	request came before any parts uploads have started.
-				 **/
-				await this.checkIfUploadCancelled(uploadId);
-
-				// Upload as many as `queueSize` parts simultaneously
-				await this.uploadParts(
-					uploadId,
-					parts.slice(start, start + this.queueSize)
+		try {
+			this.body = await this.validateAndSanitizeBody(this.params.Body);
+			this.totalBytesToUpload = this.byteLength(this.body);
+			if (this.totalBytesToUpload <= this.minPartSize) {
+				// Multipart upload is not required. Upload the sanitized body as is
+				this.params.Body = this.body;
+				const putObjectCommand = new PutObjectCommand(this.params);
+				return this.s3client.send(putObjectCommand);
+			} else {
+				// Step 1: Initiate the multi part upload
+				const uploadId = await this.createMultiPartUpload();
+
+				// Step 2: Upload chunks in parallel as requested
+				const numberOfPartsToUpload = Math.ceil(
+					this.totalBytesToUpload / this.minPartSize
 				);
 
-				/** Call cleanup a second time in case there were part upload requests
-				 *  in flight. This is to ensure that all parts are cleaned up.
-				 */
-				await this.checkIfUploadCancelled(uploadId);
-			}
+				const parts: Part[] = this.createParts();
+				for (
+					let start = 0;
+					start < numberOfPartsToUpload;
+					start += this.queueSize
+				) {
+					/** This first block will try to cancel the upload if the cancel
+					 *	request came before any parts uploads have started.
+					 **/
+					await this.checkIfUploadCancelled(uploadId);
+
+					// Upload as many as `queueSize` parts simultaneously
+					await this.uploadParts(
+						uploadId,
+						parts.slice(start, start + this.queueSize)
+					);
 
-			parts.map(part => {
-				this.removeEventListener(part);
-			});
+					/** Call cleanup a second time in case there were part upload requests
+					 *  in flight. This is to ensure that all parts are cleaned up.
+					 */
+					await this.checkIfUploadCancelled(uploadId);
+				}
 
-			// Step 3: Finalize the upload such that S3 can recreate the file
-			return await this.finishMultiPartUpload(uploadId);
+				parts.map(part => {
+					this.removeEventListener(part);
+				});
+
+				// Step 3: Finalize the upload such that S3 can recreate the file
+				return await this.finishMultiPartUpload(uploadId);
+			}
+		} catch (error) {
+			logger.error('Error. Cancelling the multipart upload. ', error.message);
+			throw error.message;
```

Comment:
```suggestion
			throw error;
```

I think we should throw the error as is. correct me if im wrong but I believe there will be no stack trace if you throw a string